comments support

yausername opened the pull request:
The code is crappy and needs lot of work.

![68747470733a2f2f692e696d6775722e636f6d2f62497a784b73702e6a7067](https://user-images.githubusercontent.com/1891273/45027275-3dbd9e80-b041-11e8-968b-4815bfa23c8f.jpg)

TobiGr commented:
@yausername Please rebase your changes in this and in the extractor repo so we can start to test and review.

theScrabi commented:
I will start to reviw this soon.

yausername commented:
requires https://github.com/TeamNewPipe/NewPipeExtractor/pull/101

yausername commented:
@TobiGr I have made some change for 

- adding content setting to disable comments.
- switch tabs with horizontal swipe - viewpager
- Changed from parallaxscrollview to coordinate layout.

Please test and give feedback if/when you have time

Thanks

TobiGr commented:
@yausername Thanks for taking a look at my suggestions. I am sorry for the late response, but I rarely find spare time for larger PRs at the moment. 

> Yes I am fetching comments asynchronously. Did you check on the latest code 803b8ea because the video detail seems to load just as quick for me.

Maybe it was my wifi. It is kinda unstable and slow...

I really like your changes so far. Comment support will be an significant improvement for NewPipe 👍 There are still some things to do from UI and UX side, but I guess someone will find time to take a look at the code soon.

Below you can see two screenshots. The left one was taken with NewPipe 0.14.2 and the right one at your latest commit. As you can see (I hope so, the divider is hard to see at my screenshots), we separate the "next video" from the other recommended videos. It might be good if you keep the next video and separate it from the others.

<p>
<img src="https://user-images.githubusercontent.com/17365767/46819960-5417f200-cd85-11e8-97b8-d65b43659a9b.png" width="300" />
<img src="https://user-images.githubusercontent.com/17365767/46819961-5417f200-cd85-11e8-9bd5-8b3b23652b71.png" width="300" />
</p>

When comments are disabled, there is no "next video" heading. But to be honest, I am not sure if we need one. @theScrabi What do you think?
We might also discuss what to do with the black divider which usually divides the tab headings and their content. In my opinion, we can keep it, but there should be more horizontal space around it.

<img src="https://user-images.githubusercontent.com/17365767/46819959-5417f200-cd85-11e8-964f-5ba5fac5acf3.png" width="300" />

theScrabi commented:
There is another thing. I don't like the huge tab 'commends' and 'downloads' as they take away a loot of screenspace. 
I'd rather like to have something like floating indicator dots: ![](https://i.imgur.com/qcEQ81R.jpg)

avently commented:
@yausername thanks, man! This is what I wanted long time

yausername commented:
@theScrabi I have added indicator dots in place of tabs.
@TobiGr I have added next video in the list. I have also added youtube style autoplay switch just updates the autoplay flag in shared preferences. We can remove it if doesn't make sense.

[Imgur](https://i.imgur.com/RLyW720.gif)

yausername commented:
![Imgur](https://i.imgur.com/RLyW720.gif)

theScrabi commented:
Hard like 👍

TobiGr commented:
@yausername Looks great! ❤️

I was missing the autoplay toggle. Awesome you added it. the toggle looks a little off-set on tablets. But maybe this can be fixed by syncing your branch.

theScrabi commented:
> I was missing the autoplay toggle.

I know that toggle is a good idea, but maybe it was better placed in the player, since it will affect the PlayerQueue.

TobiGr commented:
@theScrabi I would like to keep the toggle in the video detail fragment because people will see and use it. If you want, we can add a toggle to the queue list, too. But in my opinion, this toggle will be quite unused until this area is better accessible.
Where would you like to place it?
@yausername @karyogamy what do you think about this?

yausername commented:
@TobiGr I agree with you that the toggle will be easily accessible on the video detail fragment.
@theScrabi there is an issue. If I disable autoplay while the video is playing, the next video plays anyway.

TobiGr commented:
> there is an issue. If I disable autoplay while the video is playing, the next video plays anyway.

This will be fixed in https://github.com/TeamNewPipe/NewPipe/pull/1763

theScrabi commented:
Ok

TobiGr commented:
> > there is an issue. If I disable autoplay while the video is playing, the next video plays anyway.
>
> This will be fixed in https://github.com/TeamNewPipe/NewPipe/pull/1763

Since @theScrabi closed #1763, we should correct this in either this PR or a new one. @yausername Would you like to do this or should we open a new ticket for it?

yausername commented:
> > > there is an issue. If I disable autoplay while the video is playing, the next video plays anyway.
> > 
> > 
> > This will be fixed in #1763
> 
> Since @theScrabi closed #1763, we should correct this in either this PR or a new one. @yausername Would you like to do this or should we open a new ticket for it?

@TobiGr I will raise a separate PR for this. Should we remove all queued streams or only the last one if "autoplay/auto queuing" is disabled?

Mhowser commented:
Any update to this? Only thing keeping me from fully switching over is ability to view comments.

TobiGr commented:
> 
> 
> > > > there is an issue. If I disable autoplay while the video is playing, the next video plays anyway.
> > > 
> > > 
> > > This will be fixed in #1763
> > 
> > 
> > Since @theScrabi closed #1763, we should correct this in either this PR or a new one. @yausername Would you like to do this or should we open a new ticket for it?
> 
> @TobiGr I will raise a separate PR for this. Should we remove all queued streams or only the last one if "autoplay/auto queuing" is disabled?

I'd suggest to only remove the auto-enqueued stream (the last one). But I guess, you should wait with an implementation until #1878 is merged. That PR needs some changes and introduces the `isAutoQueued` attribute for `PlayQueueItem`s.

-----

@Mhowser There are a bunch of PRs which need a review atm. We already took a look at the UI, now we need to check the code carefully. btw @yausername, please rebase your fork to make this easier for us :)

Brimont commented:
@yausername you can build an apk debug?

TobiGr commented on app/src/main/java/org/schabi/newpipe/fragments/list/BaseListFragment.java:
```diff
@@ -181,6 +182,13 @@ public void selected(PlaylistInfoItem selectedItem) {
+        infoListAdapter.setOnCommentsSelectedListener(new OnClickGesture<CommentsInfoItem>() {
+            @Override
+            public void selected(CommentsInfoItem selectedItem) {
+                //Log.d("comments" , "this comment was clicked" + selectedItem.getCommentText());
```
I guess this can be removed or the log entry should be specified.

yausername commented:
> @yausername you can build an apk debug?

@Brimont  It is pretty old code because I haven't merged the new changes yet.
https://tinyurl.com/y86xh9g3

yausername commented:
@TobiGr Thanks for taking time. I will rebase/merge latest changes in both the PRs.

yausername commented:
@theScrabi @TobiGr I have merged latest changes here as well as in the extractor. Please take a look.
I haven't tested this on a tablet and I am pretty sure it won't work on tablets.

Brimont commented:
@yausername fantastic work apk debug work perfectly. thanks you so much

Brimont commented:
@yausername not all links on the comment are spam. YouTuber uses the comment to add the link of the products covered in the video

gkeegan commented:
Would it be possible to have a setting to disable clickable links in the comments?

Mhowser commented:
What about support for timestamps, too?

On Wed, Dec 5, 2018, 11:05 AM Keegan Given <notifications@github.com wrote:

> Would it be possible to have a setting to disable clickable links in the
> comments?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/TeamNewPipe/NewPipe/pull/1650#issuecomment-444602957>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AXEx9MUmndDVJMEAs6J3kZuOncHt9DTQks5u2Bj2gaJpZM4WXFnw>
> .
>

yausername commented:
Hi @TobiGr ,
I have removed dislike button and added published time of comment as suggested by @Mhowser 
I have also fixed tablet ui. Please test this because I could ony test it on emulator.
Thanks.

yausername commented:
@theScrabi @TobiGr @all 
Inspired by [alientube](https://github.com/xlexi/alientube) and [royt](https://github.com/mustafakalash/royt), I tried to put reddit comments on youtube videos.
Do you think this is a useful thing?

![Imgur](https://i.imgur.com/0u8d0Cn.gif)

TobiGr commented:
This sounds like an interesting feature. But I'd suggest to wait with it until we have finished this PR. I assume that this would add even more code to the extractor PR - that's more code which needs to be reviewed.
I'd like to get this merged as soon as possible, therefore we should keep both PRs small.

yausername commented:
Sure. I will wait for these PRs to be merged first.

On Sat, Dec 15, 2018, 5:38 AM Tobias Groza <notifications@github.com wrote:

> This sounds like an interesting feature. But I'd suggest to wait with it
> until we have finished this PR. I assume that this would add even more code
> to the extractor PR - that's more code which needs to be reviewed.
> I'd like to get this merged as soon as possible, therefore we should keep
> both PRs small.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/TeamNewPipe/NewPipe/pull/1650#issuecomment-447515276>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE9kP71zdk8DAoL9rXqEJ-s9d33KRTAXks5u5D1xgaJpZM4WXFnw>
> .
>

FaizKhan5 commented:
@yausername I'm using this feature right now and i notice that comments text size is too small and hard to read please increase the size.

I'm newbie

yausername commented:
> @yausername I'm using this feature right now and i notice that comments text size is too small and hard to read please increase the size and also add option to see comments reply.
> 
> I'm newbie

@FaizKhan5 sorry for the late reply.
  

- Replies will require significant changes in the extractor and ui, so it might take time.
- Personally i like smaller text size because you can fit more content in less space. But if more people think it is too small to read then we can increase it a little. Also, you can always increase font size in android settings.

FaizKhan5 commented:
0.15.0 is release now so you have some time to put replies function, but thanks for comments support. 

Spending more time  on reading  comments in small size will affect eyesight and i face that problem so that's why i want little increase in size to see comments from normal distance.

theScrabi commented:
@yausername this feature is on my agenda this year, so once we can say that 0.15.x runs rather stable again I will should try to add a next round of features.

theScrabi commented:
@TobiGr you have ben granded write access for frontend as well as backend, please feel free to review and merge PRs (especially the smaller ones). You would help me a lot, and help to make releases go faster :)

avently commented:
Finally.
@TobiGr, as your first task, please, merge this PR:) with your help this project will get new life

TobiGr commented:
@theScrabi Thanks, I'll take a look at the extractor once I finished my exam on monday.

theScrabi commented:
Merged the extractor though :). I had time today.

theScrabi commented:
Code does not look bad to me. However could someone else please give another final review?

theScrabi commented:
Please also think about QA.
Therefore here is a version build and signed with my keys:
[NewPipe-v0.15.1_with_commends.apk.zip](https://github.com/TeamNewPipe/NewPipe/files/2881076/NewPipe-v0.15.1_with_commends.apk.zip)

justanidea commented:
This works well on my device. Anyways, would be possible to set the default pannel on the next videos? I use it more often than comments, and i think its going to be really annoying to swipe at each video
A big TY anyways

avently commented:
Absolutely agree with panels reordering

yausername commented:
> ![screenshot_20190219-180434_newpipe](https://user-images.githubusercontent.com/1891273/53033370-319f8f00-3471-11e9-9e82-a6dfbf8d3507.png)
> 
> Padding in settings should be removed.

Done

theScrabi commented:
@justanidea maybe the last state of the panel could be saved.
You guys are welcome to help :)

theScrabi commented:
F*** it, it worx.

theScrabi commented:
@yausername thanks a lot, great work :+1: 

![](https://media1.giphy.com/media/l41YdHuqqelb9NJe0/giphy.gif?cid=3640f6095c6eba43506e792f6f23ea64)

cryptono commented:
@theScrabi, when is the next apk release going to be published? Can't wait for the comment support! :+1:

theScrabi commented:
Lets say next thing i do in newpipe is another release ;)

theScrabi commented:
@yausername I encountered the [first bug](https://github.com/TeamNewPipe/NewPipe/issues/2143), would you please take a look at it?

yausername commented:
> @yausername I encountered the [first bug](https://github.com/TeamNewPipe/NewPipe/issues/2143), would you please take a look at it?

Sure, I'll take a look.

skylarmt commented:
I just want to say thank you for the comments support, it's always been annoying when a video says there's more info in the comments.  However, it's funny how at the same time NewPipe gets comments, YouTube starts shutting down comments on an entire category of videos :)

Mhowser commented:
Yeah, being able to read the comments is a huge plus. Next thing we should
have is comment replies.

On Sun, Mar 3, 2019, 11:06 AM Skylar Ittner <notifications@github.com>
wrote:

> I just want to say thank you for the comments support, it's always been
> annoying when a video says there's more info in the comments. However, it's
> funny how at the same time NewPipe gets comments, YouTube starts shutting
> down comments on an entire category of videos :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/TeamNewPipe/NewPipe/pull/1650#issuecomment-469053304>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AXEx9IicL7yeHCSX--8fASuKVNLJifE4ks5vTB0xgaJpZM4WXFnw>
> .
>
